src/storage/store/column_read.cpp

```diff
@@ -0,0 +1,616 @@
+
+        auto writeToPageBufferHelper = getWriteToPageBufferHelper<InputType, T>(data, numValues);
+
+        const auto bitpackHeader = FloatCompression<T>::getBitpackInfo(state.metadata.compMeta);
+        offset_t curExceptionIdx = exceptionChunk->findFirstExceptionAtOrPastOffset(offsetInChunk);
+        for (size_t i = 0; i < numValues; ++i) {
+            const size_t writeOffset = offsetInChunk + i;
+            const size_t readOffset = srcOffset + i;
+
+            if (nullMask && nullMask->isNull(readOffset)) {
+                continue;
+            }
+
+            const T newValue = writeToPageBufferHelper.getValue(readOffset);
+            const auto& floatMetadata = metadata.compMeta.floatMetadata();
+            const int64_t encodedValue =
+                alp::AlpEncode<T>::encode_value(newValue, floatMetadata.fac, floatMetadata.exp);
+            const T decodedValue =
+                alp::AlpDecode<T>::decode_value(encodedValue, floatMetadata.fac, floatMetadata.exp);
+
+            bool newValueIsException = newValue != decodedValue;
+            writeToPageBufferHelper.setValue(i,
+                newValueIsException ? bitpackHeader.offset : newValue);
+
+            // if the previous value was an exception
+            // either overwrite it (if the new value is also an exception) or remove it
+            if (curExceptionIdx < exceptionChunk->getExceptionCount() &&
+                exceptionChunk->getExceptionAt(curExceptionIdx).posInChunk == writeOffset) {
+                if (newValueIsException) {
+                    exceptionChunk->writeException(
+                        EncodeException<T>{newValue, narrowingConversion<uint32_t>(writeOffset)},
+                        curExceptionIdx);
+                } else {
+                    exceptionChunk->removeExceptionAt(curExceptionIdx);
+                }
+                ++curExceptionIdx;
+            } else if (newValueIsException) {
+                exceptionChunk->addException(
+                    EncodeException<T>{newValue, narrowingConversion<uint32_t>(writeOffset)});
+            }
+        }
+
+        defaultReader->writeValuesToPage(state, offsetInChunk, writeToPageBufferHelper.getData(), 0,
+            numValues, writeFunc, nullMask);
+    }
+
+    std::unique_ptr<DefaultColumnReadWriter> defaultReader;
+};
+
+} // namespace
+
+template<std::floating_point T>
+InMemoryExceptionChunk<T>::InMemoryExceptionChunk(ColumnReadWriter* columnReader,
+    Transaction* transaction, const ChunkState& state)
+    : exceptionCount(state.metadata.compMeta.floatMetadata().exceptionCount),
+      finalizedExceptionCount(exceptionCount),
+      exceptionCapacity(state.metadata.compMeta.floatMetadata().exceptionCapacity),
+      exceptionBuffer(
+          std::make_unique<std::byte[]>(exceptionCapacity * EncodeException<T>::sizeBytes())),
+      emptyMask(exceptionCapacity) {
+    emptyMask.setNullFromRange(exceptionCount, exceptionCapacity - exceptionCount, true);
```

Comment:
add comment explaining that the exceptions on disk are copied to the in-memory buffer in the constructor